### PR TITLE
Adding ability to filter lifecycle tests

### PR DESCRIPTION
**Pull request by jeremyrickard**
This adds the ability to filter lifecycle tests using an environment variable (TEST_MODULES). There are three possible paths when using the variable:
1.) the developer can provide a single module name or comma delimited list of module names and only the test cases matching will be enabled.  
2.) If the variable is empty, all tests will run.
3.) If an invalid module name is provided, no tests will run.

Implements #157

### CODE REVIEW

**Comment by jeremyrickard** [comment]
Here are a couple of examples of using the environment variable to selectively run tests (some output omitted).

In this case, only run the aci and mssql tests:
```console
$ TEST_MODULES=aci,mssql make test-service-lifecycles
== RUN   TestServices
2017/12/29 18:46:50 ----> creating resource group "test-4c7deba3-b111-412e-bffd-84d6429d7695"
2017/12/29 18:46:52 ----> created resource group "test-4c7deba3-b111-412e-bffd-84d6429d7695"
2017/12/29 18:46:52 ----> testing in resource group "test-4c7deba3-b111-412e-bffd-84d6429d7695"
2017/12/29 18:46:52 Running tests for modules: [aci mssql]
=== RUN   TestServices/lifecycle
=== RUN   TestServices/lifecycle/TestServices/lifecycle/aci
=== RUN   TestServices/lifecycle/TestServices/lifecycle/mssql/new_server_and_database
=== RUN   TestServices/lifecycle/TestServices/lifecycle/mssql/database_on_an_existing_server
2017/12/29 18:46:52 ----> TestServices/lifecycle/aci: starting
2017/12/29 18:46:52 ----> TestServices/lifecycle/mssql/database_on_an_existing_server: starting
2017/12/29 18:46:52 ----> TestServices/lifecycle/mssql/new_server_and_database: starting
```


Here is an example of running with a bad module name. 
```console
$ TEST_MODULES=nomodule make test-service-lifecycles
=== RUN   TestServices
2017/12/29 18:22:54 ----> creating resource group "test-54b07116-64a0-4efc-9f63-e159b1aa4150"
2017/12/29 18:22:56 ----> created resource group "test-54b07116-64a0-4efc-9f63-e159b1aa4150"
2017/12/29 18:22:56 ----> testing in resource group "test-54b07116-64a0-4efc-9f63-e159b1aa4150"
2017/12/29 18:22:56 Running tests for modules: [nomodule]
2017/12/29 18:22:56 No test cases selected. Please check TEST_MODULES variable.
=== RUN   TestServices/lifecycle
2017/12/29 18:22:56 ----> deleting resource group "test-54b07116-64a0-4efc-9f63-e159b1aa4150"
2017/12/29 18:23:43 ----> deleted resource group "test-54b07116-64a0-4efc-9f63-e159b1aa4150"
--- PASS: TestServices (49.36s)
    --- PASS: TestServices/lifecycle (0.00s)
PASS
ok  	github.com/Azure/open-service-broker-azure/tests/lifecycle	49.379s

File: tests/lifecycle/test_case_test.go [hunk 1]
```diff
@@ -28,6 +28,10 @@ type serviceLifecycleTestCase struct {
 	testCredentials        func(credentials service.Credentials) error
 }
 
+func (s serviceLifecycleTestCase) getModule() service.Module {
```
Comment (krancour) [inline tests/lifecycle/test_case_test.go]: Do we need this function? It's not exported, so I guess only code in this same package calls it. That being the case, we may as well let would-be callers of this function access the `module` attribute of `serviceLifecycleTestCase` directly.

File: tests/lifecycle/test_cases_test.go [hunk 2]
```diff
@@ -31,13 +35,47 @@ func getTestCases(resourceGroup string) ([]serviceLifecycleTestCase, error) {
...
 	}
-
+	if len(testCases) == 0 {
+		log.Print("No test cases selected. Please check TEST_MODULES variable.")
+	}
 	return testCases, nil
 }
+
+func filter(testCases []serviceLifecycleTestCase, filters map[string]struct{}) []serviceLifecycleTestCase {
+	//If filters is empty, we are not filtering so include ethe testcase
```
Comment (krancour) [inline tests/lifecycle/test_cases_test.go]: s/ethe/all the/

File: tests/lifecycle/test_cases_test.go [hunk 3]
```diff
@@ -31,13 +35,47 @@ func getTestCases(resourceGroup string) ([]serviceLifecycleTestCase, error) {
...
 		}
 	}
-
+	if len(testCases) == 0 {
+		log.Print("No test cases selected. Please check TEST_MODULES variable.")
+	}
 	return testCases, nil
 }
+
+func filter(testCases []serviceLifecycleTestCase, filters map[string]struct{}) []serviceLifecycleTestCase {
```
Comment (krancour) [inline tests/lifecycle/test_cases_test.go]: Love the makeshift set. Nice trick.

**Comment by krancour** [comment]
After the one lint error is fixed, LGTM.

**Comment by krancour** [comment]
Addressed feedback myself since I think @jeremyrickard is out-of-pocker atm and I want to get this merged into master sooner rather than later so I can also bundle it into #187.

**Pull request merged**